src/chunkserver/network_worker_thread.cc

```diff
@@ -361,42 +378,44 @@ void worker_read_continue(csserventry *eptr) {
 		const uint16_t totalRequestBlocks =
 				(totalRequestSize + thisPartOffset + SFSBLOCKSIZE - 1) / SFSBLOCKSIZE;
 		std::vector<uint8_t> readDataPrefix;
-		eptr->messageSerializer->serializePrefixOfCstoclReadData(readDataPrefix,
-				eptr->chunkid, eptr->offset, thisPartSize);
-		packetstruct* packet = worker_create_detached_packet_with_output_buffer(readDataPrefix);
+		eptr->messageSerializer->serializePrefixOfCstoclReadData(
+		    readDataPrefix, eptr->chunkId, eptr->offset, thisPartSize);
+		PacketStruct *packet =
```

Comment:
`auto *` can be used here to avoid `PacketStruct` type.

Reply:
In this case, as the type is not implied in the context, I prefer to use the explicit type. I use it for example here:
auto *foo = new PacketStruct();